test/integration/Utils.kt

```diff
@@ -15,20 +15,156 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  *
  */
+@file:Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
 
 package com.vaticle.typedb.studio.test.integration
 
+import androidx.compose.ui.test.assertAll
+import androidx.compose.ui.test.hasClickAction
+import androidx.compose.ui.test.junit4.createComposeRule
 import androidx.compose.ui.test.junit4.ComposeContentTestRule
+import androidx.compose.ui.test.onAllNodesWithText
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import com.vaticle.typedb.client.api.TypeDBSession
+import com.vaticle.typedb.client.api.TypeDBTransaction
+import com.vaticle.typedb.studio.Studio
+import com.vaticle.typedb.studio.framework.common.WindowContext
+import com.vaticle.typedb.studio.framework.material.Icon
+import com.vaticle.typedb.studio.state.StudioState
+import com.vaticle.typedb.studio.state.common.util.Label
+import kotlinx.coroutines.delay
 import kotlinx.coroutines.runBlocking
+import java.io.File
 import java.nio.charset.StandardCharsets
 import java.nio.file.Files
+import java.nio.file.Path
 import java.nio.file.Paths
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+
+val SAVE_ICON_STRING = Icon.Code.FLOPPY_DISK.unicode
+val PLUS_ICON_STRING = Icon.Code.PLUS.unicode
+val PLAY_ICON_STRING = Icon.Code.PLAY.unicode
+val CHECK_ICON_STRING = Icon.Code.CHECK.unicode
+val ROLLBACK_ICON_STRING = Icon.Code.ROTATE_LEFT.unicode
+val CHEVRON_UP_ICON_STRING = Icon.Code.CHEVRON_UP.unicode
+val DOUBLE_CHEVRON_DOWN_ICON_STRING = Icon.Code.CHEVRONS_DOWN.unicode
+val DOUBLE_CHEVRON_UP_ICON_STRING = Icon.Code.CHEVRONS_UP.unicode
+val ARROW_FROM_SQUARE_ICON_STRING = Icon.Code.ARROW_UP_RIGHT_FROM_SQUARE.unicode
+val REFRESH_ICON_STRING = Icon.Code.ROTATE.unicode
+val XMARK_ICON_STRING = Icon.Code.XMARK.unicode
+
+val SAMPLE_DATA_PATH = File("test/data/sample_file_structure").absolutePath
+val TQL_DATA_PATH = File("test/data").absolutePath
+
+const val QUERY_FILE_NAME = "query_string.tql"
+const val DATA_FILE_NAME = "data_string.tql"
+const val SCHEMA_FILE_NAME = "schema_string.tql"
+
+const val DB_ADDRESS = "localhost:1729"
 
 fun runComposeRule(compose: ComposeContentTestRule, rule: suspend ComposeContentTestRule.() -> Unit) {
     runBlocking { compose.rule() }
 }
 
+fun studioTest(compose: ComposeContentTestRule, funcBody: suspend () -> Unit) {
+    runComposeRule(compose) {
+        setContent {
+            Studio.MainWindowContent(WindowContext(1000, 1000, 0, 0))
+        }
+        compose.waitForIdle()
+        funcBody()
+    }
+}
+
 fun fileNameToString(fileName: String): String {
     return Files.readAllLines(Paths.get(fileName), StandardCharsets.UTF_8).filter { line -> !line.startsWith('#') }
         .joinToString("")
 }
+
+fun cloneAndOpenProject(composeRule: ComposeContentTestRule, path: String, name: String): Path {
+    val absolute = File(File(name).absolutePath)
```

Comment:
We get the absolute path of the file and point another file to that. Within bazel's sandboxed execution environment, files change around the running program frequently and we want to be certain about the path that we are talking about.

To illustrate, imagine a file `File("hello.txt")`. This file doesn't have an absolute path or definite location, and exactly which 'fully qualified' file we are talking about can change based on external changes to the file system. 

For this reason, we store the absolute path of the File, calling the constructor twice.